### ISSUE TITLE
6.x erroring with `$not cannot be empty` when using `$size` operator

### ISSUE THREAD

**Issue by adamreisnz** [Automattic/mongoose#10716]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Unable to query for documents using `{someArray: {$not: {$size: 0}}}`

**If the current behavior is a bug, please provide the steps to reproduce.**
```
const mongoose = require('mongoose')
const {Schema} = mongoose
const uri = `mongodb://localhost:27017/test`
console.log(`Mongoose version: ${mongoose.version}`)

const BarSchema = new Schema({
  bar: {
    type: String,
  },
})

//Schema
const FooSchema = new Schema({
  foo: {
    type: String,
  },
  bars: [BarSchema],
})

//Model
const Foo = mongoose.model('Foo', FooSchema)

//Test function
async function test() {

  //Connect to database
  await mongoose.connect(uri)

  //Create new item
  const foo = new Foo({
    foo: 'foo',
    bars: [
      {
        bar: 'bar1',
      },
      {
        bar: 'bar2',
      },
    ],
  })
  await foo.save()

  //Query
  const foos = await mongoose
    .model('Foo')
    .find({bars: {$not: {$size: 0}}})

  console.log(foos)
}

test()
```

**What is the expected behavior?**
To find the document:

![image](https://user-images.githubusercontent.com/490562/133148660-67626a4f-8b28-4783-b4fd-4bc87e3757c4.png)

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Mongoose 6.0.5
MongoDB shell version v4.2.8
NodeJS 16.3.0

This was working ok in Mongoose 5.x

**Comment by vkarpov15** [comment]
Fixed by #10730, fix will be in 6.0.7 :+1:

**Comment by adamreisnz** [comment]
Thank you
